xds/src/main/java/io/grpc/xds/XdsClientImpl.java

```diff
@@ -649,13 +667,15 @@ private String processRouteConfig(RouteConfiguration config) {
     // hostname in original "xds:" URI.
     if (targetVirtualHost != null) {
       // The client will look only at the last route in the list (the default route),
-      // whose match field must be empty and whose route field must be set.
+      // whose match field must contain a prefix field whose value is empty string
+      // and whose route field must be set.
```

Comment:
Not sure about the spec. If the best matching `targetVirtualHost` doesn't satisfy this condition, but the rank 2 matching satisfies, still return null?

Reply:
Hmm... That's very subtle edge case. It's not specified (and I guess this detail won't be specified precisely even in the future, as this detail is very deep in terms of implementation). But I think returning null would make more sense. Domain matching decides which virtual host to be routed to, if that virtual host is not configured correctly for gRPC, then returning null as for result not found will at least raise our attention if this causes failure. If we've ever hit such an edge case, discussions will be brought up. I think for now, we don't need to consider this too complicatedly.

Reply:
It may worth a comment to describe the behavior of our implementation or rationale of our choice.

Reply:
Okay.